### PR TITLE
Add annotation to Swift Proxy LoadBalancer Service to pick correct MetalLB IP pool

### PR DESCRIPTION
**Pull request by katrybacka**

### CODE REVIEW

File: pkg/apis/contrail/v1alpha1/swiftproxy_types.go [hunk 1]
```diff
@@ -29,17 +29,12 @@ type SwiftProxyConfiguration struct {
 	RingConfigMapName     string       `json:"ringConfigMapName,omitempty"`
 	Containers            []*Container `json:"containers,omitempty"`
 	SwiftServiceType      serviceType  `json:"swiftServiceType,omitempty"`
+	// MetalLB pool name that should be used by Service
+	IPPoolName string `json:"ipPoolName,omitempty"`
```
Comment (kaweue) [inline pkg/apis/contrail/v1alpha1/swiftproxy_types.go:33]: Is it too specific? That we create metalb field in this api? Maybe we should create some sub structure like
```
type Service struct {
type 
annotations
}
```
That might be more general not limited to metallb

Comment (katrybacka) [reply pkg/apis/contrail/v1alpha1/swiftproxy_types.go:33]: so you want to pass annotations specified by the user?

File: pkg/apis/contrail/v1alpha1/swiftproxy_types.go [hunk 2]
```diff
@@ -78,12 +73,32 @@ type SwiftProxyList struct {
...
+// GetServiceAnnotations prepares annotations for LoadBalancer Service
+func (s *SwiftProxy) GetServiceAnnotations() map[string]string {
+	if s.GetServiceType() == corev1.ServiceTypeLoadBalancer {
+		return map[string]string{}
+	}
+	poolName := "mgmt"
+	if s.Spec.ServiceConfiguration.IPPoolName != "" {
+		poolName = s.Spec.ServiceConfiguration.IPPoolName
+	}
+	return map[string]string{"metallb.universe.tf/address-pool": poolName}
```
Comment (kaweue) [inline pkg/apis/contrail/v1alpha1/swiftproxy_types.go:99]: Please add TODO to remove this default value

Comment (katrybacka) [reply pkg/apis/contrail/v1alpha1/swiftproxy_types.go]: done

File: pkg/apis/contrail/v1alpha1/swiftproxy_types.go (hunk 1 above)
Comment (katrybacka) [reply pkg/apis/contrail/v1alpha1/swiftproxy_types.go]: done

**Review by kaweue — APPROVED** [review]

**Review by kabart — APPROVED** [review]

**Pull request merged**